From 282b01cb5c4eb1fdba026dac486957b61d0ae8cb Mon Sep 17 00:00:00 2001
From: Piyush Kumar <piyushkumg@gmail.com>
Date: Mon, 15 Mar 2021 14:20:49 +0530
Subject: [PATCH] Revert "use lfence instead of rep and nop"

the specific patch is a perf improvement on SKX specifically since pause
(rep nop) is artificially expensive there.

outside of SKX it's a no-op basically

This reverts commit dc676829e2e3c72276c840189a5ed4c6be319e97.
---
 arch/x86/include/asm/vdso/processor.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/arch/x86/include/asm/vdso/processor.h b/arch/x86/include/asm/vdso/processor.h
index e2c45674f989..57b1a7034c64 100644
--- a/arch/x86/include/asm/vdso/processor.h
+++ b/arch/x86/include/asm/vdso/processor.h
@@ -10,7 +10,7 @@
 /* REP NOP (PAUSE) is a good thing to insert into busy-wait loops. */
 static __always_inline void rep_nop(void)
 {
-	asm volatile("lfence" ::: "memory");
+	asm volatile("rep; nop" ::: "memory");
 }
 
 static __always_inline void cpu_relax(void)
